src/westpa/analysis/trajectories.py

```diff
@@ -255,6 +256,48 @@ def fget(walker, include_initpoint=True, atom_indices=None, sim_root='.'):
         self.segment_collector.show_progress = True
 
 
+class HDF5MDTrajectory(Trajectory):
+    """A MD trajectory stored by the HDF5 framework.
+
+    Parameters
+    ----------
+    atom_indices : list, default `None`
+
+    """
+
+    def __init__(self, atom_indices=None):
+        def fget(walker, include_initpoint=False, atom_indices=atom_indices):
+            from westpa.core.h5io import WESTIterationFile
+
+            iter = walker.iteration
+            if 'trajectories' not in iter.h5group:
+                raise KeyError("The loaded west.h5 does not seem to be generated by the HDF5 framework")
```

Comment:
From the user's perspective, does it make sense for this to be a `KeyError`? I'm wondering if maybe it should be a `ValueError`.